pyanaconda/modules/payloads/source/url/url.py

```diff
@@ -0,0 +1,145 @@
+from pyanaconda.modules.common.errors import InvalidValueError
+from pyanaconda.modules.common.structures.payload import RepoConfigurationData
+from pyanaconda.modules.payloads.constants import SourceType, URLType
+from pyanaconda.modules.payloads.source.source_base import PayloadSourceBase
+from pyanaconda.modules.payloads.source.url.url_interface import URLSourceInterface
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+class URLSourceModule(PayloadSourceBase):
+    """The URL source payload module."""
+
+    REPO_NAME_ID = 0
+
+    def __init__(self):
+        super().__init__()
+        self._repo_configuration = RepoConfigurationData()
+        self.repo_configuration_changed = Signal()
+
+        self._url_source_id = self._generate_source_id()
+        self._set_repo_config_name()
+
+        self._install = False
+        self.install_changed = Signal()
+
+    @classmethod
+    def _generate_source_id(cls):
+        source_id = cls.REPO_NAME_ID
+        cls.REPO_NAME_ID = cls.REPO_NAME_ID + 1
+
+        return source_id
+
+    def _set_repo_config_name(self):
+        self._repo_configuration.name = "{}-{}".format(BASE_REPO_NAME, self._url_source_id)
+
+    def is_ready(self):
+        """This source is ready for the installation to start."""
+        # FIXME: always true is correct but it will block change of payload source. Find solution!
+        return True
+
+    @property
+    def type(self):
+        """Get type of this source."""
+        return SourceType.URL
+
+    def for_publication(self):
+        """Get the interface used to publish this source."""
+        return URLSourceInterface(self)
+
+    def set_up_with_tasks(self):
+        """Set up the installation source.
+
+        :return: list of tasks required for the source setup
+        :rtype: [Task]
+        """
+        return []
+
+    def tear_down_with_tasks(self):
+        """Tear down the installation source.
+
+        :return: list of tasks required for the source clean-up
+        :rtype: [Task]
+        """
+        return []
+
+    @property
+    def repo_configuration(self):
+        """Get repository configuration data.
+
+        :rtype: RepoConfigurationData data structure
+        """
+        return self._repo_configuration
+
+    def set_repo_configuration(self, repo_configuration):
+        """Set repository configuration data.
+
+        :param repo_configuration: configuration for this repository
+        :type repo_configuration: RepoConfigurationData data structure
+        """
+        self._repo_configuration = repo_configuration
+
+        self._validate_url(self._repo_configuration.type)
+        self._validate_proxy(self._repo_configuration.proxy)
+
+        if not self._repo_configuration.name:
+            self._set_repo_config_name()
+
+        self.repo_configuration_changed.emit(self._repo_configuration)
+        log.debug("The repo_configuration has changed")
```

Comment:
Please, log the configuration. You can simply write:
`log.debug("The repo configuration is set to: %s", repo_configuration)`

Reply:
Fixed.